extension/extension.go

```diff
@@ -59,7 +60,7 @@ type StatusWatcher interface {
 	// Extensions that implement this interface must be ready that the ComponentStatusChanged
 	// may be called before, after or concurrently with calls to Component.Start() and Component.Shutdown().
 	// The function may be called concurrently with itself.
-	ComponentStatusChanged(source *component.InstanceID, event *component.StatusEvent)
+	ComponentStatusChanged(source *component.InstanceID, event *componentstatus.StatusEvent)
 }
```

Comment:
Should this interface also be in componentstatus?

Reply:
I think that would make sense.  Since that would be a breaking change, I think it makes sense to add it in the PR that moves stuff to `componentstatus`, and deprecate the one in `service/extensions`.